### PR TITLE
Further MainActivity optimizations

### PR DESCRIPTION
**Pull request by SchoolGuy**
This PR tries to fix the following things:

- `MainApplication.java`: Remove an additional white space which I missed in a previous PR.
- `MainActivty.java`:
    - Remove a method which was only calling an intent in OTHER classes. Calling an Intent is nothing I would deduplicate if it only happens twice. The import of `MainActivtiy.java` was removed and the code was placed in the appropiate places.
    - Inlined a single usage of a method which calls an Intent.
    - Converted an anonymous object into a lambda function.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
@bekuno Split out from the original... How is this PR looking for you?

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java [hunk 1]
```diff
@@ -96,8 +97,9 @@ public void onCreate(Bundle savedInstanceState) {
             loc1.setLongitude(MainActivity.cartridgeFile.longitude);
             Guide guide = new Guide(MainActivity.cartridgeFile.name, loc1);
             A.getGuidingContent().guideStart(guide);
-            MainActivity.callGudingScreen(CartridgeDetailsActivity.this);
-            CartridgeDetailsActivity.this.finish();
+            Intent intent = new Intent(CartridgeDetailsActivity.this, GuidingActivity.class);
+            startActivity(intent);
+            finish();
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:102]: Please shorten in the same way in line 90.

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java [hunk 2]
```diff
@@ -687,13 +675,6 @@ private void clearPackageFromMemory() {
         }).start();
     }
 
-    private void clickMap() {
-        MapDataProvider mdp = MapHelper.getMapDataProvider();
-        mdp.clear();
-        mdp.addCartridges(cartridgeFiles);
-        MainActivity.wui.showScreen(WUI.SCREEN_MAP, null);
-    }
-
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java:696]: Please let this as own method. I would say that it is so better readable.

**Review by bekuno — CHANGES_REQUESTED** [review]
Additional:
The cleanup in `PreferenceValues` in PR #240 would be better fit in this PR.

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java (hunk 2 above)
Comment (SchoolGuy) [reply src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java:696]: I am sometimes amazed by what people find readable... Jumping 200 lines for four lines of code is an interesting definition of readable. But you are the boss...

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java (hunk 1 above)
Comment (SchoolGuy) [reply src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:102]: Can we please write all of the style rules down or enforce them. I find line 102 logical, even more in the light of what I have done in line 101. I personally find line 91 somehow illogical as I am literally just calling a method of an object. Prefixing that is just a wast of characters imho. But again. You are the boss here.

**Comment by SchoolGuy** [comment]
@bekuno I addressed all of your desired changes I think.

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java (hunk 1 above)
Comment (bekuno) [reply src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java]: I really meant `shorten` - in L91.
To `finish();` as you had done in L102.

Comment (SchoolGuy) [reply src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:102]: Ahhh okay that makes a lot of sense. Sorry for the misunderstanding. Correction is done!

**Review by bekuno — APPROVED** [review]

**Pull request merged**